src/types.ts

```diff
@@ -26,31 +26,28 @@ export interface Spec<T> {
   devDefault?: NonNullable<T> | undefined
 }
 
-type OptionalSpec<T> = Omit<Spec<T>, 'default'> & { default: undefined }
+type OptionalFlag<T> =
```

Comment:
`Flag` is a little confusing to me here. Would it make sense to change the suffix to `Attrs`, both here and for `RequiredFlag`?